src/main/java/se/sundsvall/parkingpermit/integration/casedata/mapper/CaseDataMapper.java

```diff
@@ -27,43 +27,43 @@
-			.collect(Collectors.toCollection(ArrayList::new));
+		// DisplayStatus is set if provided, or to an empty list if absent
+		ofNullable(nullableDisplayPhase).ifPresentOrElse(
+			displayPhase -> extraParameters.add(new ExtraParameter(CASEDATA_KEY_DISPLAY_PHASE).values(List.of(displayPhase))),
+			() -> extraParameters.add(new ExtraParameter(CASEDATA_KEY_DISPLAY_PHASE).values(emptyList())));
 
-		Optional.ofNullable(displayPhase).ifPresentOrElse(display -> result.add(new ExtraParameter(CASEDATA_KEY_DISPLAY_PHASE).values(List.of(display))),
-			() -> result.add(new ExtraParameter(CASEDATA_KEY_DISPLAY_PHASE).values(emptyList())));
-
-		return patchErrand.extraParameters(result);
+		return extraParameters;
 	}
 
-	public static PatchErrand toPatchErrand(final String externalCaseId, final String phase, final String phaseStatus, final String phaseAction, final List<ExtraParameter> extraParameters) {
-		final var patchErrand = new PatchErrand()
-			.externalCaseId(externalCaseId)
-			.phase(phase);
-
-		var result = Optional.ofNullable(extraParameters).orElse(emptyList()).stream()
-			.filter(extraParameter -> !CASEDATA_KEY_PHASE_STATUS.equals(extraParameter.getKey()) && !CASEDATA_KEY_PHASE_ACTION.equals(extraParameter.getKey()))
-			.collect(Collectors.toCollection(ArrayList::new));
+	public static List<ExtraParameter> toExtraParameterList(final String nullablePhaseStatus, final String nullablePhaseAction) {
+		final var extraParameters = new ArrayList<ExtraParameter>();
 
-		Optional.ofNullable(phaseStatus).ifPresentOrElse(status -> result.add(new ExtraParameter(CASEDATA_KEY_PHASE_STATUS).values(List.of(status))),
-			() -> result.add(new ExtraParameter(CASEDATA_KEY_PHASE_STATUS).values(emptyList())));
+		// PhaseStatus is set if provided, or to an empty list if absent
+		ofNullable(nullablePhaseStatus).ifPresentOrElse(
+			phaseStatus -> extraParameters.add(new ExtraParameter(CASEDATA_KEY_PHASE_STATUS).values(List.of(phaseStatus))),
+			() -> extraParameters.add(new ExtraParameter(CASEDATA_KEY_PHASE_STATUS).values(emptyList())));
 
-		// Cannot be null since that could erase action when it is "AUTOMATIC"
-		Optional.ofNullable(phaseAction).ifPresentOrElse(action -> result.add(new ExtraParameter(CASEDATA_KEY_PHASE_ACTION).values(List.of(action))),
+		// PhaseAction cannot be null since that could erase action when it is "AUTOMATIC"
+		ofNullable(nullablePhaseAction).ifPresentOrElse(phaseAction -> extraParameters.add(new ExtraParameter(CASEDATA_KEY_PHASE_ACTION).values(List.of(phaseAction))),
 			() -> { throw new IllegalArgumentException("phaseAction cannot be null"); });
 
-		return patchErrand.extraParameters(result);
+		return extraParameters;
+	}
+
+	public static PatchErrand toPatchErrand(final String externalCaseId, final String phase) {
+		return new PatchErrand()
+			.externalCaseId(externalCaseId)
+			.labels(null) // Set this explicitly to null to not update labels to empty list when patching errand
```

Comment:
There is a config option for this:
https://github.com/OpenAPITools/openapi-generator/blob/master/docs/generators/java.md

`containerDefaultToNull`